assembl/static2/js/app/components/common/termsForm.jsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import { Translate } from 'react-redux-i18n';
+import { Modal, Button } from 'react-bootstrap';
+import { connect } from 'react-redux';
+import { compose, graphql } from 'react-apollo';
+import { closeModal } from '../../utils/utilityManager';
+import LegalNoticeAndTerms from '../../graphql/LegalNoticeAndTerms.graphql';
+import withLoadingIndicator from '../../components/common/withLoadingIndicator';
+
+class TermsForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isScrolled: false
+    };
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentDidMount() {
+    const wrappedElement = document.getElementById('box');
+    wrappedElement.addEventListener('scroll', (event) => {
+      const element = event.target;
+      if (element.scrollHeight - element.scrollTop === element.clientHeight) {
+        this.setState({
+          isScrolled: true
+        });
+      }
+    });
+  }
+
+  handleSubmit() {
+    this.props.handleAcceptButton();
+    closeModal();
+  }
+
+  render() {
+    const { isScrolled } = this.state;
+    const { isChecked, text } = this.props;
+    return (
+      <div className="terms-form">
+        <Modal.Header closeButton>
+          <Modal.Title>
+            <Translate value="termsAndConditions.headerTitle" />
+          </Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <div className="terms-box justify" id="box" dangerouslySetInnerHTML={{ __html: text }} />
```

Comment:
remove the `box` id if you don't need it

Reply:
I do need it to target this element with document.getElementById
But maybe there is a better way to do that

Reply:
will refacto this part

Reply:
it's ok if you use it, maybe rename it to `terms-and-conditions-box` to ensure there is only one on the page

Reply:
I used ref instead, see last refactoring comment :)

Reply:
much better!